docs/data/data-grid/components/JoyUISlots.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { DataGridPro } from '@mui/x-data-grid-pro';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import joySlots from '@mui/x-data-grid/joy';
+import type {} from '@mui/material/themeCssVarsAugmentation';
+import {
+  extendTheme as extendJoyTheme,
+  shouldSkipGeneratingVar as joyShouldSkipGeneratingVar,
+} from '@mui/joy/styles';
+import {
+  Experimental_CssVarsProvider as CssVarsProvider,
+  experimental_extendTheme as extendMuiTheme,
+  shouldSkipGeneratingVar as muiShouldSkipGeneratingVar,
+} from '@mui/material/styles';
+import { deepmerge } from '@mui/utils';
+import { blue, grey } from '@mui/material/colors';
+
+const { unstable_sxConfig: muiSxConfig, ...muiTheme } = extendMuiTheme();
```

Comment:
I've followed https://mui.com/joy-ui/guides/using-joy-ui-and-material-ui-together/#case-a-joy-ui-in-a-material-ui-project here.
Not sure how to make it work with the docs mode switch though:

https://user-images.githubusercontent.com/13808724/221645237-10a5da34-daca-45cd-a49a-ab6f163071bd.mov